modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -817,89 +873,148 @@ class PrePostProcWrapper {
         }
     }
 
+    void cfgColorFormat(const std::string &input_name) {
+        auto &input_info = m_ppp.input(input_name);
+        input_info.tensor().set_color_format(::ov::preprocess::ColorFormat::NV12_TWO_PLANES, {"y", "uv"});
+    }
+
     // FIXME: Decompose this...
     void cfgPreProcessing(const std::string  &input_name,
                           const cv::GMetaArg &input_meta,
                           const bool         disable_img_resize = false) {
-        GAPI_Assert(cv::util::holds_alternative<cv::GMatDesc>(input_meta));
-        const auto &matdesc = cv::util::get<cv::GMatDesc>(input_meta);
+        GAPI_Assert(cv::util::holds_alternative<cv::GMatDesc>(input_meta) ||
+                    cv::util::holds_alternative<cv::GFrameDesc>(input_meta));
+        if (cv::util::holds_alternative<cv::GMatDesc>(input_meta)) {
```

Comment:
> The point is that when we have GFrameDesc is should always fall into isImage category, since MediaFrame is type to represent an image so this function can be extended to accept GMetaArg and if it holds GFrameDesc returns true otherwise asserts it's GMatDesc and fall to already existing implementation.

That's what I kept for `MediaFrame`. If you take a look at the `else if (cv::util::holds_alternative<cv::GFrameDesc>(input_meta)) {` branch you'll see the logic that aligns with `isImage` branch from `cv::GMatDesc` case.

Reply:
Yes, but can we extend `isImage` to handle `GFrameDesc` as well? In that case we wouldn't need to duplicate logic inside `if`.

Reply:
Done, thanks! https://github.com/opencv/opencv/commit/24e5ceb635e0f70b6b5302f077bab9d5c315a12b